pkg/api/nextdate.go

```diff
@@ -0,0 +1,90 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const DateFormat = "20060102"
+
+func NextDate(now time.Time, dstart string, repeat string) (string, error) {
+	if repeat == "" {
+		return "", fmt.Errorf("repeat rule is empty")
+	}
+
+	startDate, err := time.Parse(DateFormat, dstart)
+	if err != nil {
+		return "", fmt.Errorf("error with start date: %w", err)
+	}
+
+	parts := strings.Split(repeat, " ")
+
+	switch parts[0] {
+	case "d":
+		if len(parts) != 2 {
+			return "", fmt.Errorf("error with format for date rule")
+		}
+		interval, err := strconv.Atoi(parts[1])
+		if err != nil || interval < 1 || interval > 400 {
+			return "", fmt.Errorf("error with interval for date rule")
+		}
+
+		date := startDate
+		for {
+			date = date.AddDate(0, 0, interval)
+			if date.After(now) {
+				return date.Format(DateFormat), nil
+			}
+		}
+
+	case "y":
+
+		next := startDate
+		for {
+			next = next.AddDate(1, 0, 0)
+			if next.After(now) {
+				return next.Format(DateFormat), nil
+			}
+		}
+
+	case "w":
+
+		return "", fmt.Errorf("w rule not implemented")
+
+	case "m":
+		return "", fmt.Errorf("m rule not implemented")
+
+	default:
+		return "", fmt.Errorf("unsupported rule: %s", repeat)
+	}
+}
+
+func nextDayHandler(w http.ResponseWriter, r *http.Request) {
+	nowStr := r.FormValue("now")
+	dstart := r.FormValue("date")
+	repeat := r.FormValue("repeat")
+
+	var now time.Time
+	var err error
+
+	if nowStr == "" {
+		now = time.Now()
+	} else {
+		now, err = time.Parse(DateFormat, nowStr)
+		if err != nil {
+			http.Error(w, "error with now date", http.StatusBadRequest)
+			return
+		}
+	}
+
+	next, err := NextDate(now, dstart, repeat)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	fmt.Fprint(w, next)
```

Comment:
Лучше не игнорировать ошибки
```suggestion
	_, err = fmt.Fprint(w, next)
```